app.rb

```diff
@@ -0,0 +1,177 @@
+require_relative './modules/preserver_module'
+require_relative './modules/book_module'
+require_relative './modules/music_album_module'
+require_relative './modules/game_module'
+require_relative './classes/books'
+require_relative './classes/label'
+require_relative './classes/music_album'
+require_relative './classes/genre'
+require_relative './classes/game'
+
+class App
+  include PreserverModule
+  include BookModule
+  include MusicModule
+  include GameModule
+  attr_reader :books
+
+  def initialize
+    @books = []
+    @albums = []
+    @authors = []
+    @genres = []
+    @labels = []
+    @sources = []
+    @games = []
+
+    load_data
+  end
+
+  def start_program
+    puts "\n    Welcome to the catalog app  :)"
+    until list_of_options
+      input = gets.chomp
+      if input == '0'
+        preserve_files
+        puts "\n    Thank you for using our app  :) \n "
+        break
+      end
+      option(input)
+    end
+  end
+
+  def list_all_books
+    puts 'No available books' if @books.empty?
+    @books.each { |book| puts "#{book['publisher']} #{book['publish_date']} #{book['cover_state']}" }
+  end
```

Comment:
[OPTIONAL] It would be nice if you also included the book `name` and `title` when listing books.

- That means you might have to include the `name` variable in your Book class in order to be able to create book names when adding a new book.